build_tools/pytorch.py

```diff
@@ -68,7 +68,7 @@ def setup_pytorch_extension(
         print("Could not determine CUDA Toolkit version")
     else:
         if version >= (11, 2):
-            nvcc_flags.extend(["--threads", "4"])
+            nvcc_flags.extend(["--threads", os.getenv("NVTE_BUILD_THREADS_PER_JOB", "1")])
```

Comment:
@phu0ngng could you document here the reason for switching from 4 to 1 as a default?

Reply:
As @timmoon10  mentioned, users getting stuck in the build process (https://github.com/NVIDIA/TransformerEngine/issues/976, https://github.com/NVIDIA/TransformerEngine/issues/978, https://github.com/NVIDIA/TransformerEngine/issues/1077). 
Besides, the PyTorch Ext build with `--thread 4` broke on my desktop. I discussed with Tim and we think it is better to have 1 as default so that all users have fewer build issues.
For "advanced" users, they are encouraged to play with `NVTE_BUILD_THREADS_PER_JOB` to best match their system.